src/server/internalServer.cpp

```diff
@@ -137,10 +138,24 @@ InternalServer::InternalServer(Library* library,
   m_withTaskbar(withTaskbar),
   m_withLibraryButton(withLibraryButton),
   m_blockExternalLinks(blockExternalLinks),
+  m_customIndexTemplate(customIndexTemplate),
   mp_daemon(nullptr),
   mp_library(library),
   mp_nameMapper(nameMapper ? nameMapper : &defaultNameMapper)
-{}
+{
+  if (!m_customIndexTemplate.empty()) {
+    m_customIndexTemplate = kiwix::isRelativePath(m_customIndexTemplate) ?
+                              computeAbsolutePath(getCurrentDirectory(), m_customIndexTemplate) :
+                              m_customIndexTemplate;
+    m_indexTemplateString = getFileContent(m_customIndexTemplate);
+    if (m_indexTemplateString.empty() && m_verbose.load()) {
+      printf("Invalid path for customIndexTemplate, falling back to default template.\n");
+    }
+  }
```

Comment:
I'm not sure we simply want to print a message and go with(out) it.

If a user ask to use a specific template and we cannot find it, it is probably a error.

---

In fact, I wonder if the user may directly pass the template string to the constructor and not the path to a file. This simplify a lot the things on libkiwix side and allow use case where the template is not stored in a file (generated template, ...)
It would be to kiwix-serve/kiwix-desktop/wrapper to load the template and pass it to the server's constructor.

Reply:
@mgautierfr any example or some resource I can refer for this....not exactly a cpp dev.
I guess we should have a discussion regarding this with @kelson42 before going ahead.

Reply:
You have pretty everything you need.
Instead of adding a argument `customIndexTemplate(path)`, directly add the argument `indexTemplateString`. (And use `RESOURCE::templates::index_html` if the string is empty)

And move the code loading the template from `customIndexTemplate` (you 've already implemented) in kiwix-serve.

But yes, we may agree with @kelson42 on the feature before going ahead.